### PR TITLE
added tests for CVSB-18921

### PR DESCRIPTION
**Pull request by DeepikaSinghD**
This PR contains automated test cases for CVSB-18921 https://jira.dvsacloud.uk/browse/CVSB-18921 for VTG5A and VTG30 test types for HGV and TRL

### CODE REVIEW

File: src/test/java/trailerRegistration/PostTrailerRegistration.java [hunk 1]
```diff
@@ -265,4 +276,277 @@ public void testTrailerUnRegistrationForIntReasonForDeRegistration() {
...
+        JsonPathAlteration alterationTestResult = new JsonPathAlteration("$.testTypes[0].testResult", "fail","","REPLACE");
+
+        List<JsonPathAlteration> TestResultAlterations = new ArrayList<>(Arrays.asList(
+                alterationTestResultVin,
+                alterationTestResultId,
+                alterationSystemNumber,
+                alterationTestResult
+        ));
+
+        //testResultsSteps.waitForTestResultsToBePosted(testResultRecord, TestResultAlterations,5);
```
Comment (IuliaDVSA) [inline src/test/java/trailerRegistration/PostTrailerRegistration.java:492]: Please remove commented out code

**Review by IuliaDVSA — CHANGES_REQUESTED** [review]
Please remove commented code line 492

File: src/test/java/trailerRegistration/PostTrailerRegistration.java (hunk 1 above)
Comment (DeepikaSinghD) [reply src/test/java/trailerRegistration/PostTrailerRegistration.java]: Done

File: src/test/java/trailerRegistration/PostTrailerRegistration.java [hunk 2]
```diff
@@ -265,4 +276,275 @@ public void testTrailerUnRegistrationForIntReasonForDeRegistration() {
...
     }
+
+    @Title("CVSB-18921 - AC1 -  VTG5A TRN PRESENT - TRL Annual Test Pass - With TRN")
+    @Test
+    public void testPostTestResultPassTrailerRegistrationCertificateWithTrn() {
+
+        // Read the base test result JSON
+        String postTechnicalRecord = GenericData.readJsonValueFromFile("technical-records_trl_all_fields.json","$");
+
+        //String randomVin = GenericData.generateRandomVin();
```
Comment (bhavia-dvsa) [inline src/test/java/trailerRegistration/PostTrailerRegistration.java:287]: can you remove the unwanted codes please (the commented codes are there in all tests)

File: src/test/java/trailerRegistration/PostTrailerRegistration.java [hunk 3]
```diff
@@ -265,4 +276,275 @@ public void testTrailerUnRegistrationForIntReasonForDeRegistration() {
...
+        List<JsonPathAlteration> alterations = new ArrayList<>(Arrays.asList(
+                alterationVin));
+
+        vehicleTechnicalRecordsSteps.postVehicleTechnicalRecordsWithAlterations(postTechnicalRecord, alterations);
+        vehicleTechnicalRecordsSteps.statusCodeShouldBe(HttpStatus.SC_CREATED);
+        vehicleTechnicalRecordsSteps.validateData("Technical Record created");
+        vehicleTechnicalRecordsSteps.getVehicleTechnicalRecordsByStatus(randomVin, VehicleTechnicalRecordStatus.ALL);
+        vehicleTechnicalRecordsSteps.statusCodeShouldBe(200);
+        String systemNumber = vehicleTechnicalRecordsSteps.getSystemNumber();
+
```
Comment (IuliaDVSA) [inline src/test/java/trailerRegistration/PostTrailerRegistration.java:300]: Please removed extra blank line

File: src/test/java/trailerRegistration/PostTrailerRegistration.java [hunk 4]
```diff
@@ -265,4 +276,275 @@ public void testTrailerUnRegistrationForIntReasonForDeRegistration() {
...
+    }
+
+    @Title("CVSB-18921 - AC3 - VTG30 TRN PRESENT - TRL Annual Test Fail - With TRN")
+    @Test
+    public void testPostTestResultFailTrailerRegistrationCertificateWithTrn() {
+
+        // Read the base test result JSON
+        String postTechnicalRecord = GenericData.readJsonValueFromFile("technical-records_trl_all_fields.json","$");
+
+        //String randomVin = GenericData.generateRandomVin();
```
Comment (IuliaDVSA) [inline src/test/java/trailerRegistration/PostTrailerRegistration.java:397]: Please remove unnecessary code.

**Review by IuliaDVSA — CHANGES_REQUESTED** [review]
A couple of more comments

**Review by bhavia-dvsa — APPROVED** [review]
approving the tests for CVSB-18921

**Review by IuliaDVSA — APPROVED** [review]
All sorted

**Pull request merged**